scene/2d/tile_map_layer.cpp

```diff
@@ -2565,6 +2657,105 @@ void TileMapLayer::set_enabled(bool p_enabled) {
 bool TileMapLayer::is_enabled() const {
 	return enabled;
 }
+void TileMapLayer::set_tileset(const Ref<TileSet> &p_tileset) {
+	if (p_tileset == tile_set) {
+		return;
+	}
+
+	dirty.flags[DIRTY_FLAGS_TILE_SET] = true;
+	_queue_internal_update();
+	emit_signal(CoreStringNames::get_singleton()->changed);
```

Comment:
Shouldn't this emit after changing? Otherwise you'll get outdated data if you do something with this signal.